### ISSUE TITLE
Remove yellow border from full-screen window

### ISSUE THREAD

**Issue by somatronicon** [rohatsu/ShaderGlass#11]
First off, great project! Absolutely love it. Until people figure out there is a market for poly lenses that emulate CRTs for LED monitors (akin to this: https://hackaday.com/2021/06/16/fit-your-lcds-with-lenses-for-that-vintage-crt-look/), I'm using ShaderGlass 👍 

As I'm trying to immerse myself in the past, I'm a bit distracted by the yellow border that appears around the full-screen window. My configuration can be seen below:

```
ProfileVersion "1.0"
PixelSize "1"
AspectRatio "1"
ShaderCategory "crt"
ShaderName "crt-lottes-multipass"
FrameSkip "2"
OutputScale "100"
FlipH "0"
FlipV "0"
Clone "0"
CaptureCursor "1"
Transparent "1"
```

And the screenshot:

![image](https://user-images.githubusercontent.com/86700194/123897580-4174eb00-d929-11eb-9a0c-891a140e22bd.png)

**Comment by somatronicon** [comment]
Sorry, just noticed that the screenshot above doesn't show the shaders being applied. However, you can still see the yellow border around the ShaderGlass main window.

**Comment by rohatsu** [comment]
Hey, thanks for the feedback. Yes the yellow border is unfortunate but it's a security feature of Windows Capture API and cannot be removed (OBS etc. also has this problem). The only way to avoid it is to use clone mode (either capture the other monitor if you have multiple, or capture just the window of your game/app and put ShaderGlass in front of it), please try setting up your content that way.

**Comment by somatronicon** [comment]
Very interesting... Totally understandable then as it's a limitation of the API used to capture. I'll give clone a shot and see what happens. Unfortunately I only have a single monitor, and am after retro-izing my entire desktop... So, I might just have to live with the yellow border for now.

**Comment by somatronicon** [comment]
BTW, I just stumbled onto this; maybe useful?:

https://www.gitmemory.com/issue/microsoft/Windows.UI.Composition-Win32-Samples/46/829032545

**Comment by somatronicon** [comment]
Also, just for some context, I traced that snippet back to the following github issue; unfortunately, I can't tell if it'd be applicable in this case judging from the API docs:

https://github.com/microsoft/Windows.UI.Composition-Win32-Samples/issues/46

**Comment by trebory6** [comment]
So I was going through some Windows documentation, and found that this page now references a borderless capture requests. 

Maybe this can help solve this issue? I'll drop it here just in case. 

https://docs.microsoft.com/en-us/uwp/api/windows.graphics.capture.graphicscapturesession.isborderrequired?view=winrt-20348#Windows_Graphics_Capture_GraphicsCaptureSession_IsBorderRequired

**Comment by rohatsu** [comment]
Great find @trebory6 , I will test this out!

**Comment by greysondn** [comment]
Hi.

Was hoping to write a full screen LUT shader for reasons, stumbled headlong into this issue "yet again", just popping a note here to say I'd love to see it solved.

Otherwise, thanks for your application; it's been a trip down memory lane testing it against old game footage. I've not felt this young in... well, my age minus about 8 years ;)

**Comment by shinra358** [comment]
obs does not have this problem

**Comment by ner00** [comment]
True, I think this is the relevant portion: https://github.com/obsproject/obs-studio/pull/4070

**Comment by Hey** [comment]
> obs does not have this problem

It does if you use the "Window 10 (1903 and up)" display capture method.

**Comment by nakami** [comment]
Any updates on this issue?

**Comment by mausimus** [comment]
I am not able to reproduce yellow border on latest Windows 11 (22H2), I've added steps to confirm ShaderGlass has Windows' permission to suppress it to the [FAQ](https://github.com/mausimus/ShaderGlass/blob/master/FAQ.md) but in my testing they get enabled by default. If anyone is on latest Windows 11 and those steps don't make yellow border disappear, please post here.

**Comment by QUAKEULUS** [comment]
Still an issue on
Microsoft Windows 10 Enterprise
10.0.19045 N/A Build 19045

with ShaderGlass-0.9-win-x64

**Comment by Tasosgemah** [comment]
Still an issue in Windows 10 pro 22H2

**Comment by dengdwc** [comment]
it can be gone if you got 2 screen , and keep turning the f11 project option to some point. i did  it with a projector  :O

**Comment by ner00** [comment]
I project to a second screen and then switch back to the original screen and that works, but unfortunately this only works if you have a second screen/display output connected.

**Comment by RedZedda** [comment]
Created a new account just for this. I noticed that whenever I had left to go outside, my monitors had blanked out like normal. When I went to turn my game back on, the border was COMPLETELY gone. I'm on windows 10 and I just wanted to see if anyone else knew why this would have happened. Like the border was gone until I closed everything out and went to bed.

**Comment by ner00** [comment]
@RedZedda Interesting. I was able to reproduce this by... powering off/on my monitor lmao

**Comment by Tasosgemah** [comment]
I can't get rid of it even with turning the monitor on/off trick.

**Comment by ner00** [comment]
I'm not sure then. For me it works, but for context:
- I have 2 monitors, 1 through DisplayPort and 1 through HDMI;
- 2nd monitor is plugged but turned off and not detected by the OS while turned off;
- I open ShaderGlass (only using 1st monitor) and get the yellow border;
- I poweroff the 1st monitor and power it on again, the yellow border is no longer present.

I don't know if there is some residual OS acknowledgement of the 2nd monitor, even though it is powered off since the beginning. All I know is that it cleared the yellow border by powering off and powering on the main monitor.

**Comment by Tasosgemah** [comment]
> I'm not sure then. For me it works, but for context:
> 
>     * I have 2 monitors, 1 through DisplayPort and 1 through HDMI;
> 
>     * 2nd monitor is plugged but turned off and not detected by the OS while turned off;
> 
>     * I open ShaderGlass (only using 1st monitor) and get the yellow border;
> 
>     * I poweroff the 1st monitor and power it on again, the yellow border is no longer present.
> 
> 
> I don't know if there is some residual OS acknowledgement of the 2nd monitor, even though it is powered off since the beginning. All I know is that it cleared the yellow border by powering off and powering on the main monitor.

Did the exact steps (i have the same stuff as you) and i managed to replicate this. The yellow line does indeed go away.

Unfortunately, this also freezes the shaderglass window. Meaning, it gets stuck to the image before i turned off the monitor and won't update. So it's unusable.

**Comment by ner00** [comment]
@Tasosgemah My bad for not checking properly. You're absolutely correct, it will get stuck in the last frame of whatever window/process it was "shading", and thus unusable. Even my comment from last year about projecting to a second monitor and back achieves the same stale result or doesn't remove the border at all. Seems like Windows 11 is the only way to go, which is a pity because I'm not migrating to Windows 11 ever lol

**Comment by Tasosgemah** [comment]
Yeah, me neither. Shader Glass is cool but it's not worth dealing with Windows 11 for now.

**Comment by Chaseroni** [comment]
Same thing here and with shader beam, really would like to make these go away easily.
